### DISCUSSION TITLE
Using the k parameter with STUMPI

### DISCUSSION THREAD

**Discussion by jimacDS** [TDAmeritrade/stumpy#796] [Help: Coding & Implementations]
On the documentation for stumpi, there is a parameter **k**, that is supposed to be able to set:

 "The number of top k smallest distances used to construct the matrix profile. Note that this will increase the total computational time and memory usage when k > 1."

However, when trying to set k=2, I receive the error:

`TypeError: aampi.__init__() got an unexpected keyword argument 'k'`

Has this been deprecated

**Comment by seanlaw** [comment]
@jimacDS Welcome to the STUMPY community and thank you for your question. 

> Has this been deprecated

Actually, it's the opposite. Our docs usually reflect any changes in our development branch and so `k` is currently only available if you install STUMPY from source. This parameter was only added recently and will be available in our upcoming release.

In the meantime, you can install from source by doing something like:

```
$ python -m pip install git+https://github.com/TDAmeritrade/stumpy.git
```

Sorry for the confusion!

**Reply by jimacDS** [reply]
Thanks Sean!

**Reply by tholds** [reply]
@seanlaw, I was also confused by the API docs (coincidentally, when trying to pass _k_ to _stumpy.stump_). There are a number of issues which might be worth looking at:

1. [stumpy.readthedocs.io](https://stumpy.readthedocs.io) redirects to [stumpy.readthedocs.io/en/latest/](https://stumpy.readthedocs.io/en/latest/index.html) and there is no mention of, or link to, the stable release documentation [https://stumpy.readthedocs.io/en/stable/](https://stumpy.readthedocs.io/en/stable/). 
2. The [page title](https://www.w3schools.com/tags/tag_title.asp) for [stumpy.readthedocs.io/en/latest/](https://stumpy.readthedocs.io/en/latest/index.html) refers to the stable version, 1.11.1 (see image below)
3. This is inconsistent with other packages (notably [NumPy](https://numpy.org/), [Pandas](https://pandas.pydata.org/) and [SciPy](https://docs.scipy.org/doc/scipy/), which direct users to docs for the stable version by default, clearly identify the relevant version and provide a way to select other versions.
4. Both the README and the [stable docs](https://stumpy.readthedocs.io/en/stable/), provide install instructions for the stable release (pip / conda), but include hyperlinks to documentation that direct the user to the latest docs (potentially inconsistent with the version they have installed). See [index.html#how-to-use-stumpy](https://stumpy.readthedocs.io/en/stable/index.html#how-to-use-stumpy) for an example.


<img width="544" alt="image" src="https://github.com/TDAmeritrade/stumpy/assets/41975630/13c956a5-bf0d-4891-ac1f-324a7e604113">

**Reply by seanlaw** [reply]
@tholds Apologies for the inconvenience. Would you be willing to help us sort this out? I haven't had an opportunity to think through how to best manage/correct this behavior and we could use the help. I guess "stable" should really be tied to a precise release and, maybe, there is a way to hide/remove "latest".
